src/components/MoneyTransaction.js

```diff
@@ -7,13 +7,12 @@ import { Heading, HeadingDisabled } from './Text'
 import { db } from '../firebase-config'
 import { collection, getDocs, addDoc } from 'firebase/firestore'
 
-export const MoneyTransaction = ({ user }) => {
+export const MoneyTransaction = ({ user = [] }) => {
```

Comment:
Here you could have extracted the array to a constant, or even better define a `const DEFAULT_EMPTY_ARRAY = []` in a dedicated file and use that one everywhere